concepts/functions/about.md

```diff
@@ -6,60 +6,85 @@ Python, like other programming languages, has [_built-in functions_][build-in fu
-
-A function can also return a value using the `return` keyword. The value returned by the function is known as the `return value`. The return value is returned to the caller of the function.
+To execute the code inside a function, you need to call the function, which is done by using the function name followed by parenthesese [`()`].
+Data, known as [`arguments`][arguments], can be passed to the function by placing them inside the parenthesese.
+A function definition may include zero or more [`parameters`][parameters].
+Parameters define what argument(s) the function accepts.
+A parameter defines what _type_ of argument is to be passed.
+The argument is the actual _value_ passed when the function is called.
+Functions can perform different tasks depending on the arguments passed to the parameters.
+
+A function can also return a value using the [`return`][return] keyword.
+The value returned by the function is known as the `return value`.
+The return value is returned to the caller of the function.
 
 ## Creation
 
-In python, functions are created using the `def` keyword. The function definition is followed by the function name and parenthesis [`()`]. Inside the parenthesis, the parameters are specified, separated by commas. After the close parenthesis, the colon (`:`) is used to separate the function signature from the function body.
+In python, functions are created using the [`def`][def] keyword.
+The function definition is followed by the function name and parentheses [`()`].
+Inside the parentheses, the parameters are specified, separated by commas.
+After the close parenthesis, the colon (`:`) is used to separate the function signature from the function body.
 
-The function body is a block of code that is executed when the function is called. The body of the function is indented. The indentation is important because Python relies on it to know where that block of code ends. A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
+The function body is a block of code that is executed when the function is called.
+The body of the function is indented.
+The indentation is important because Python relies on it to know where that block of code ends.
+A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
 
 ```python
 def function_name(parameter1, parameter2, ...):
     # function body
-    return value
+    return parameter1 + parameter2
+
 ```
 
 We can also define a function without any parameters or return value.
 
 ```python
 def function_name():
     # function body
+    pass
+
 ```
 
+Note that the function does need a body, even if the body does nothing, or trying to run the program will generate an indentation error.
```

Comment:
```suggestion
Note that the function does need a body, even if the body does nothing, or trying to run the program will generate an indentation error:

```python
>>> def my_bodyless_func():
...      

File ~/temp.py:1
    
    ^
IndentationError: expected an indented block
```
The keyword `pass` can be used in the function body to avoid the error.
```